data_structures/bloom_filters/bloom_filter/bloom_filter.c

```diff
@@ -0,0 +1,161 @@
+        h ^= murmur_32_scramble(k);
+        h = (h << 13) | (h >> 19);
+        h = h * 5 + 0xe6546b64;
+    }
+    /* Read the rest. */
+    k = 0;
+    for (unsigned long i = len & 3; i; i--)
+    {
+        k <<= 8;
+        k |= key[i - 1];
+    }
+    // A swap is *not* necessary here because the preceding loop already
+    // places the low bytes in the low places according to whatever endianness
+    // we use. Swaps only apply when the memory is copied in a chunk.
+    h ^= murmur_32_scramble(k);
+    /* Finalize. */
+    h ^= len;
+    h ^= h >> 16;
+    h *= 0x85ebca6b;
+    h ^= h >> 13;
+    h *= 0xc2b2ae35;
+    h ^= h >> 16;
+    return h;
+}
+
+struct BloomFilter
+{
+    int hashFuncs;  ///< The number of hash functions this bloom filter will use
+    int numSlots;   ///< The number of buckets in the underlying array.
+    int maxElems;   ///< The expected max number of elements to be inserted
+    unsigned char buckets[];  ///< The underlying array structure of the filter
+};
+
+void bloom_printStats(struct BloomFilter *this)
+{
+    int numOnes = 0;
+    for (int i = 0; i < this->numSlots; ++i)
+    {
+        int slot = i / 8;
+        int bit = i % 8;
+        if ((this->buckets[(int)slot] >> bit) == 1)
+        {
+            ++numOnes;
+        }
+    }
+
+    float nStar = -((float)this->numSlots / (float)this->hashFuncs) *
+                  log(1.0 - ((float)numOnes / (float)this->numSlots));
+
+    printf(
+        "Max Elements (n): %d \nHash Functions (k): %d \nSlots (m): "
+        "%d\nEstimated Elements Inserted (n*): %f (estimate deteriorates as "
+        "inserted elements approaches n)\n",
+        this->maxElems, this->hashFuncs, this->numSlots, nStar);
+}
+
+void bloom_insert(struct BloomFilter *this, void *elem, size_t size)
+{
+    for (int i = 0; i < this->hashFuncs; ++i)
+    {
+        int spot = murmur3_32((uint8_t *)elem, size, i) % this->numSlots;
+        int slot = spot / 8;
+        int bit = spot % 8;
+        this->buckets[(int)slot] |= (1 << bit);
+    }
+    return;
+}
+
+enum bloomResponse bloom_contains(struct BloomFilter *this, void *elem,
+                                  size_t size)
+{
+    for (int i = 0; i < this->hashFuncs; ++i)
+    {
+        int spot = murmur3_32((uint8_t *)elem, size, i) % this->numSlots;
+        int slot = spot / 8;
+        int bit = spot % 8;
+        if ((this->buckets[(int)slot] >> bit) == 0)
+        {
+            return DEFINITELY_NOT_PRESENT;
+        }
+    }
+    return PROBABLY_PRESENT;
+}
+
+void delete_bloomFilter(struct BloomFilter *this) { free(this); }
+
+struct BloomFilter *construct_bloomFilter(int expectedElems,
+                                          float falsePositiveProb)
+{
+    int m = -floor((expectedElems * log(falsePositiveProb)) / pow(log(2), 2));
+    int k = ceil((m / expectedElems) * log(2));
+
+    struct BloomFilter *new = calloc(
+        1, sizeof(struct BloomFilter) + sizeof(unsigned char) * ((m + 1) / 8));
```

Comment:
Maybe you should note that you are using calloc() to initialize all bits to 0, people might get confused why are you calling calloc() with 1